ee/server/service/mdm.go

```diff
@@ -1195,3 +1179,194 @@ func (svc *Service) GetMDMManualEnrollmentProfile(ctx context.Context) ([]byte,
 	// need arises.
 	return mobileConfig, nil
 }
+
+func (svc *Service) SaveABMToken(ctx context.Context, token io.Reader) (*fleet.ABMToken, error) {
+	// first check for reads as we need to load the cert/key from the db. We will
+	// do another write check below.
+	if err := svc.authz.Authorize(ctx, &fleet.AppleBM{}, fleet.ActionRead); err != nil {
+		return nil, err
+	}
+
+	pair, err := svc.ds.GetAllMDMConfigAssetsByName(ctx, []fleet.MDMAssetName{
+		fleet.MDMAssetABMCert,
+		fleet.MDMAssetABMKey,
+	})
+	if err != nil {
+		if fleet.IsNotFound(err) {
+			return nil, ctxerr.Wrap(ctx, &fleet.BadRequestError{
+				Message: "Please generate a keypair first.",
+			}, "saving ABM token")
+		}
+
+		return nil, ctxerr.Wrap(ctx, err, "retrieving stored ABM assets")
+	}
+
+	tokenBytes, err := io.ReadAll(token)
+	if err != nil {
+		return nil, ctxerr.Wrap(ctx, err, "reading token bytes")
+	}
+
+	derCert, _ := pem.Decode(pair[fleet.MDMAssetABMCert].Value)
+	if derCert == nil {
+		return nil, ctxerr.Wrap(ctx, err, "ABM certificate in the database cannot be parsed")
```

Comment:
You can't call `Wrap` here as there is no `err` (it will return `nil, nil`), you need `ctxerr.New`.